### DISCUSSION TITLE
can pgagroal handle postgres pod replacement?

### DISCUSSION THREAD

**Discussion by jobcespedes** [pgagroal/pgagroal#131] [Q&A]
Hi,

I was wondering if app connections would fail or not when using pgagroal (session) as sidecar and postgres pod is deleted and recreated.

**Comment by jesperpedersen** [comment]
So you want existing sessions to keep their state ?

**Comment by jobcespedes** [comment]
Yes. No disruption from the app perspective, actually.

**Reply by jesperpedersen** [reply]
Yeah, there is still work needed in this area.

Basically it requires that the client does a login transparently using the information from `pgagroal_users.conf`.

At the moment you will have to do `disable`, `flush-` and wait for the clients to finish. Restart. Then let the clients reconnect.

Let me know if you want to work on it, and I can guide you through it !

**Comment by jobcespedes** [comment]
Thanks for the details and help. I'm just considering the options, for the moment.
